components/ShoppingListEditor/ShoppingListEditor.js

```diff
@@ -2,55 +2,99 @@ import styled from "styled-components";
 import { useRef, useState, useEffect } from "react";
 
 import ListContainer from "../ListContainer";
+import InputDropDown from "./InputDropDown";
+import { getAllItemsFromDB } from "/services/db.js";
 
 export default function ShoppingListEditor({ items, onDelete, onAdd }) {
+  const [allItems, setAllItems] = useState(getAllItemsFromDB);
+  const [itemInput, setItemInput] = useState("");
+  const [dropDownItems, setDropDownItems] = useState([]);
+  const [dropDownReset, setDropDownReset] = useState(false);
   const inputRef = useRef();
 
+  //set focus on item input after page load
   useEffect(() => {
     inputRef.current.focus();
-  }, []); //set focus on item input after page load
+  }, []);
 
-  function submitForm(event) {
-    event.preventDefault();
-    const form = event.target;
-    const formData = new FormData(form);
-    const itemName = Object.fromEntries(formData).item;
+  //open drop down when typing into input field
+  useEffect(() => {
+    triggerDropDown();
+  }, [itemInput]);
 
-    const trimmedItemName = itemName.trim(); //remove white spaces at beginning and end of the name
+  //reset dropDownItems state, to remove dropDown buttons from screen,...
+  //...when shopping list (items) changed
+  useEffect(() => {
+    setItemInput("");
+    setDropDownItems([]);
+  }, [items]);
 
-    // --- Check user input ---
-    if (trimmedItemName === "") {
-      alert("Type an item name!");
-      return;
+  //...when loosing focus of input field
+  //-> setDropDownReset state needed to allow clicking drop down buttons,
+  //which also removes focus on input field
+  useEffect(() => {
+    if (dropDownReset) {
+      setDropDownItems([]);
+      setDropDownReset(false);
     }
-    if (
-      items.find(
-        (item) => item.name.toLowerCase() === trimmedItemName.toLowerCase()
-      ) !== undefined
-    ) {
-      alert("Item already on the list!");
-      return;
+  }, [dropDownReset]);
+
+  //evoke rendering drop down buttons for items that match input
+  function triggerDropDown() {
+    const matchedDropDownItems = matchInput(itemInput);
+    setDropDownItems(matchedDropDownItems);
+  }
+
+  //match input with all available items from DB
+  function matchInput(value) {
+    const editedValue = value.trim().toLowerCase(); //trim() -> remove white spaces at beginning and end of the name
+    //clear drop down when clearing input field
+    if (editedValue === "") {
+      return [];
     }
+    const matchedItems = allItems.filter((item) =>
+      item.name.toLowerCase().startsWith(editedValue)
+    );
+    //filter out matched items, that are already on the shopping list
+    const uniqueMatchedItems = matchedItems.filter((matchedItem) => {
+      return (
+        items.find((item) => matchedItem.id === item.item_id) === undefined
```

Comment:
instead of using this find hack you can use "indexOf"
```js
return items.indexOf(matchedItem) === -1
```
indexOf gives you the index of the first occurrence of an entity in an array. If it doesn't find any (what you want) it returns -1.

Reply:
Also you might want to filter out all used items from allItems in the first step, since then they already used items don't even match your string in the first place. 
1. reduce the set which you want to search in 
2. perform the search